lib/ts/recipe/totp/recipe.ts

```diff
@@ -0,0 +1,243 @@
+            {
+                id: CREATE_DEVICE_API,
+                disabled: this.apiImpl.createDevicePOST === undefined,
+                method: "post",
+                pathWithoutApiBasePath: new NormalisedURLPath(CREATE_DEVICE_API),
+            },
+            {
+                id: VERIFY_DEVICE_API,
+                disabled: this.apiImpl.verifyDevicePOST === undefined,
+                method: "post",
+                pathWithoutApiBasePath: new NormalisedURLPath(VERIFY_DEVICE_API),
+            },
+            {
+                id: VERIFY_CODE_API,
+                disabled: this.apiImpl.verifyCodePOST === undefined,
+                method: "post",
+                pathWithoutApiBasePath: new NormalisedURLPath(VERIFY_CODE_API),
+            },
+            {
+                id: REMOVE_DEVICE_API,
+                disabled: this.apiImpl.removeDevicePOST === undefined,
+                method: "post",
+                pathWithoutApiBasePath: new NormalisedURLPath(REMOVE_DEVICE_API),
+            },
+            {
+                id: LIST_DEVICE_API,
+                disabled: this.apiImpl.listDevicesGET === undefined,
+                method: "get",
+                pathWithoutApiBasePath: new NormalisedURLPath(LIST_DEVICE_API),
+            },
+            {
+                id: IS_TOTP_ENABLED_API,
+                disabled: this.apiImpl.isTotpEnabledGET === undefined,
+                method: "get",
+                pathWithoutApiBasePath: new NormalisedURLPath(IS_TOTP_ENABLED_API),
+            },
+        ];
+    };
+
+    handleAPIRequest = async (
+        id: string,
+        req: BaseRequest,
+        res: BaseResponse,
+        _: NormalisedURLPath,
+        __: HTTPMethod
+    ): Promise<boolean> => {
+        const options = {
+            config: this.config,
+            recipeId: this.getRecipeId(),
+            isInServerlessEnv: this.isInServerlessEnv,
+            recipeImplementation: this.recipeInterfaceImpl,
+            req,
+            res,
+            appInfo: this.getAppInfo(),
+        };
+        if (id === CREATE_DEVICE_API) {
+            return await createDeviceAPI(this.apiImpl, options);
+        } else if (id == VERIFY_DEVICE_API) {
+            return await verifyDeviceAPI(this.apiImpl, options);
+        } else if (id == VERIFY_CODE_API) {
+            return await verifyCodeAPI(this.apiImpl, options);
+        } else if (id == REMOVE_DEVICE_API) {
+            return await removeDeviceAPI(this.apiImpl, options);
+        } else if (id == LIST_DEVICE_API) {
+            return await listDevicesAPI(this.apiImpl, options);
+        }
+
+        return false;
+    };
+
+    handleError = async (err: STError, _: BaseRequest, res: BaseResponse): Promise<void> => {
+        if (err.fromRecipe === TotpRecipe.RECIPE_ID) {
+            if (err.type === TotpError.TOTP_NOT_ENABLED_ERROR) {
+                sendNon200ResponseWithMessage(res, "TOTP is not enabled for the user", 400); // bad req
```

Comment:
400 is not a good resposne for this. Search for some other status code maybe

Reply:
403 (Forbidden) seems better